graphql_compiler/compiler/metadata.py

```diff
@@ -24,6 +24,13 @@
 )
 
 
+OutputInfo = namedtuple(
+    'OutputInfo',
+    (
+        'location',
```

Comment:
Let's include the type and whether the output is optional as well?

The QueryMetadataTable object should arguably be a superset of all other metadata, so in principle if we do `record_output_info()` on line 321 in the file above, we probably shouldn't need to also record this info in `context['outputs'][output_name]` as well -- instead, that should be autogenerated from the `OutputInfo` object from the QueryMetadataTable.

Reply:
I think the idea that it should be a superset makes sense. 

What do you mean by autogenerated? (that `context['outputs']` should be generated from `context['metadata']`?) Either way, why have `context['outputs']` if all the information is in `context['metadata']`? 

Generally, is the idea of context to store local information that we need for making decisions? If so, is there any problem with storing the info twice?

Reply:
Yes, I think we want to remove `context['outputs']` if that information is in the `QueryMetadataTable`. We don't need to do it in this PR, but it'd be nice to work in that direction by adding the output and optional information in the table.

Reply:
Agreed with @bojanserafimov. The `context` predates the `QueryMetadataTable` by a long time, and is how we stored query-level metadata before we had this newer, more structured approach. Over time, I expect we'll remove many if not most parts of the `context` dict.